src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -50,6 +51,67 @@ void AppendLines (string prefix, List<string> lines, StringBuilder sb)
 			}
 		}
 
+		public static int RunProcess (string command, string arguments, TaskLoggingHelper log, DataReceivedEventHandler? onOutput = null, DataReceivedEventHandler? onError = null)
+		{
+			var stdout_completed = new ManualResetEvent (false);
+			var stderr_completed = new ManualResetEvent (false);
+			var psi = new ProcessStartInfo () {
+				FileName = command,
+				Arguments = arguments,
+				UseShellExecute = false,
+				RedirectStandardOutput = true,
+				RedirectStandardError = true,
+				CreateNoWindow = true,
+				WindowStyle = ProcessWindowStyle.Hidden,
+			};
+
+			var stdoutLines = new List<string> ();
+			var stderrLines = new List<string> ();
+
+			log.LogDebugMessage ($"Running process: {psi.FileName} {psi.Arguments}");
+			using var proc = new Process ();
+			proc.OutputDataReceived += (s, e) => {
+				if (e.Data != null) {
+					onOutput?.Invoke (s, e);
+					stdoutLines.Add (e.Data);
+				} else
+					stdout_completed.Set ();
+			};
+
+			proc.ErrorDataReceived += (s, e) => {
+				if (e.Data != null) {
+					onError?.Invoke (s, e);
+					stderrLines.Add (e.Data);
+				} else
+					stderr_completed.Set ();
+			};
+
+			proc.StartInfo = psi;
+			proc.Start ();
+			proc.BeginOutputReadLine ();
+			proc.BeginErrorReadLine ();
+			proc.WaitForExit ();
+
+			if (psi.RedirectStandardError) {
+				stderr_completed.WaitOne (TimeSpan.FromSeconds (30));
+			}
+
+			if (psi.RedirectStandardOutput) {
+				stdout_completed.WaitOne (TimeSpan.FromSeconds (30));
+			}
+
+			if (proc.ExitCode != 0) {
+				var sb = MergeStdoutAndStderrMessages (stdoutLines, stderrLines);
+				log.LogError ($"Command '{psi.FileName} {psi.Arguments}' failed.\n{sb.ToString ()}");
```

Comment:
I think there should be a *slight* re-think on this, as this will be an uncoded error message, if and when it happens.

"Better" might be to do one of two things:

 1. Add a `string taskCode` parameter, allowing us to `log.LogCodedError(taskCode, …)`, *or*
 2. Replace the `TaskLoggingHelper` parameter with an `AndroidTask` parameter, allowing this method access to `.Log` and `.TaskPrefix`.  We'd still have to "concoct" an error code -- `${task.TaskPrefix}0000`? -- but it's better than no code at all, as error codes are logged via telemetry.